### PR TITLE
Add vault csi documentation

### PR DESCRIPTION
**Pull request by hanzala1234**

### CODE REVIEW

**Comment by stakater-user** [comment]
@hanzala1234 Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:SNAPSHOT-PR-235-6360620e`

File: content/sre/secrets/vault-csi.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# Vault CSI Provider
```
Comment (rasheedamir) [inline content/sre/secrets/vault-csi.md]: @hanzala1234 @sabkat this document is hard to follow; are we suggesting multiple ways one can use it?

also can we place make it more like step by step guide?

can we add a supporting diagram which explains the whole workflow?

can we add auto restart as well?

**Review by rasheedamir — CHANGES_REQUESTED** [review]
Plz look at comments

File: content/sre/secrets/vault-csi.md (hunk 1 above)
Comment (hanzala1234) [reply content/sre/secrets/vault-csi.md]: "this document is hard to follow; are we suggesting multiple ways one can use it"

we are not suggesting multiple way. we are covering different use cases

**Comment by stakater-user** [comment]
@hanzala1234 Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:SNAPSHOT-PR-235-1cf7de23`

File: content/sre/secrets/vault-csi.md (hunk 1 above)
Comment (rasheedamir) [reply content/sre/secrets/vault-csi.md]: what ever is the workflow it should be super easy for developers to follow

**Comment by stakater-user** [comment]
@hanzala1234 Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:SNAPSHOT-PR-235-0227577a`

**Comment by stakater-user** [comment]
@rasheedamir Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:SNAPSHOT-PR-235-e6dc32c0`

**Comment by stakater-user** [comment]
@rasheedamir Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:SNAPSHOT-PR-235-531d6e10`

**Comment by stakater-user** [comment]
@rasheedamir Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:SNAPSHOT-PR-235-f59832ef`

**Comment by stakater-user** [comment]
@rasheedamir Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:SNAPSHOT-PR-235-5a516ad0`

**Comment by stakater-user** [comment]
@rasheedamir Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:SNAPSHOT-PR-235-30100376`

**Comment by stakater-user** [comment]
@rasheedamir Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:SNAPSHOT-PR-235-077e12ff`

**Comment by stakater-user** [comment]
@rasheedamir Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:SNAPSHOT-PR-235-54353a08`

**Comment by stakater-user** [comment]
@hanzala1234 Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:SNAPSHOT-PR-235-dbb3fb23`

**Comment by stakater-user** [comment]
@rasheedamir Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:SNAPSHOT-PR-235-2ad543c5`

**Comment by stakater-user** [comment]
@rasheedamir Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:SNAPSHOT-PR-235-d62e9a90`

**Comment by stakater-user** [comment]
@rasheedamir Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:SNAPSHOT-PR-235-511bbb95`

**Comment by stakater-user** [comment]
@rasheedamir Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:SNAPSHOT-PR-235-1f2f3856`

**Comment by stakater-user** [comment]
@rasheedamir Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:SNAPSHOT-PR-235-bae2b038`

**Comment by stakater-user** [comment]
@rasheedamir Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:SNAPSHOT-PR-235-00257c64`

**Review by rasheedamir — APPROVED** [review]

**Comment by stakater-user** [comment]
@rasheedamir Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:v0.0.173`
